File: das_backend/src/main/java/ch/sbb/backend/logging/domain/service/DomainLoggingService.java
package ch.sbb.backend.logging.domain.service;

import ch.sbb.backend.logging.domain.model.LogEntry;
import ch.sbb.backend.logging.domain.repository.LoggingRepository;
import java.util.List;

public class DomainLoggingService implements LoggingService {

    private final LoggingRepository loggingRepository;

    public DomainLoggingService(LoggingRepository loggingRepository) {
        this.loggingRepository = loggingRepository;
    }

    @Override
    public void saveAll(List<LogEntry> logEntries) {
        loggingRepository.saveAll(logEntries);
    }
}

File: backend/src/main/kotlin/ch/sbb/backend/logging/domain/service/DomainLoggingService.kt
package ch.sbb.backend.logging.domain.service

import ch.sbb.backend.logging.domain.LogEntry
import ch.sbb.backend.logging.domain.repository.LoggingRepository

class DomainLoggingService(
    private val loggingRepository: LoggingRepository
) : LoggingService {
    override fun saveAll(logEntries: List<LogEntry>) {
        loggingRepository.saveAll(logEntries)
    }
}
